src/internal/feature/feature.go

```diff
@@ -0,0 +1,213 @@
+	// Since is the version a feature is first introduced in alpha stage.
+	Since `json:"since,omitempty"`
+	// Until is the version when a deprecated feature is fully removed. Historical versions included.
+	Until `json:"until,omitempty"`
+	Stage `json:"stage,omitempty"`
+}
+
+// IsEnabled allows users to optimistically check for a feature. Useful for control flow. Any user-enabled or disabled
+// features take precedence over the default setting.
+func IsEnabled(name Name) bool {
+	f, err := Get(name)
+	if err != nil {
+		// We don't actually need to check the error here because an empty f will have the same truthiness
+		return false
+	}
+	return bool(f.Enabled)
+}
+
+// Set takes a slice of one or many flags, inserting the features onto user-configured features. If a feature name is
+// provided that is already a part of the set, then Set will return an error.
+func Set(features []Feature) error {
+	// Ensure user features haven't been set
+	m := AllUser()
+	if len(m) > 0 {
+		return fmt.Errorf("user features have already been set")
+	}
+	userFeatures.Store(featuresToMap(features))
+	return nil
+}
+
+// setDefault takes a slice of one or many flags, inserting the features onto the default feature set. If
+// a feature name is provided that is already a part of the set, then SetDefault will return an error. This function
+// can only be called once.
+func setDefault(features []Feature) error {
+	// Ensure default features haven't been set
+	m := AllDefault()
+	if len(m) > 0 {
+		return fmt.Errorf("default features have already been set")
+	}
+	defaultFeatures.Store(featuresToMap(features))
+	return nil
+}
+
+// Get takes a flag Name and returns the Feature struct. If the doesn't exist then it will error. It will check both the
+// default set and the user set, and if a flag exists in both it will return the user data for it.
+func Get(name Name) (Feature, error) {
+	// Get from user set
+	fu, uErr := GetUser(name)
+	if uErr == nil {
+		return fu, nil
+	}
+
+	// Fallback to default set
+	fd, dErr := GetDefault(name)
+	if dErr == nil {
+		return fd, nil
+	}
+
+	// Feature not found in either set
+	return Feature{}, fmt.Errorf("feature not found: %s", name)
+}
+
+// GetDefault takes a flag Name and returns the Feature struct from the default set.
+func GetDefault(name Name) (Feature, error) {
+	f, ok := AllDefault()[name]
+	if !ok {
+		return f, fmt.Errorf("feature not found: %s", name)
+	}
+	return f, nil
+}
+
+// GetUser takes a flag Name and returns the Feature struct from the user set.
+func GetUser(name Name) (Feature, error) {
+	f, ok := AllUser()[name]
+	if !ok {
+		return f, fmt.Errorf("feature not found: %s", name)
+	}
+	return f, nil
+}
+
+// All returns all flags from both Default and User.
+func All() map[Mode]map[Name]Feature {
+	m := make(map[Mode]map[Name]Feature)
+	m[Default] = AllDefault()
+	m[User] = AllUser()
+	return m
+}
+
+// AllDefault returns all features from the Default set for this version of Zarf.
```

Comment:
Should this intentionally return the same map instance stored in `defaultFeatures` or should it be making/return a copy such that someone doesn't unintentionally (or otherwise) break the WORM guarantee and mutate a flag?